das/worker.go

```diff
@@ -0,0 +1,86 @@
+package das
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"go.uber.org/multierr"
+)
+
+type worker struct {
+	lock  sync.Mutex
+	state workerState
+}
+
+// workerState contains important information about the state of a
+// current sampling routine.
+type workerState struct {
+	Curr uint64 `json:"curr"`
+	From uint64 `json:"from"`
+	To   uint64 `json:"to"`
+
+	failed []uint64
+	Err    error `json:"error,omitempty"`
+}
+
+type job struct {
+	id         int
+	from, to   uint64
+	isPriority bool
```

Comment:
How much of a performance boost will this bring/what are your arguments for why it is important? I am more on the side of keeping things as simple as they need to be, and I am unsure if this prioritization is worth the lines of code

Reply:
Biggest benefit from introducing `priority` is the ability to prioritise recently found headers over aged ones. They are in much higher demand and needs to be propagated through network as fast as possible.

Reply:
Main discussion: resolved
Sidenote: change json curr -> current

Reply:
also why have the struct name Curr? Especially if the json is now current